streaming/base/shared/prefix.py

```diff
@@ -184,29 +210,37 @@ def get_shm_prefix(streams_local: list[str],
     # Check my locals for overlap.
     _check_self(streams_local)
 
+    prefix_int = max([
+        _check_and_find_retrying(streams_local, streams_remote, shm_name=shm_name, retry=retry)
```

Comment:
Just for my learning, why do we need retry here? is there any exception that not captured inside `_check_and_find`?

Reply:
@karan6181 may know better why it was there. 
My understanding is that there can be some racing when multiple processes try to attach the same shm file at the same time.

Reply:
There may be some old processes attempting to clean up while the new process tries to attach to shared memory. The retries allow the old processes to finish their tasks.